test/test_sparse.py

```diff
@@ -1023,6 +1023,52 @@ def test_shape(i_shapes, v_shapes, nnzs):
         test_shape([0, 3, 4], [3, 4, 5, 6], [0])
         test_shape([2, 3, 4], [0, 4, 5, 6], [9, 12])
 
+    def _test_narrow(self, input, narrow_args, dense=None):
+        if dense is None:
+            dense = input.to_dense()
+        expected = dense.narrow(*narrow_args)
+        self.assertEqual(expected, input.narrow_copy(*narrow_args).to_dense())
+
+    def test_narrow(self):
+        input = self.SparseTensor(
+            self.IndexTensor([[0], [1], [2]]).transpose(1, 0),
+            self.ValueTensor([3, 4, 5]),
+            torch.Size([3]))
+
+        narrow_args = [0, 0, 2]
```

Comment:
can you test for all valid combinations of args, e.g., [dim, start, length]? Also test for invalid inputs, .e.g, dim < 0, or dim >= input._sparseDims + input._denseDims, or start < 0, start >= size[dim], etc. Also maybe use `_gen_sparse` to get a more generalized test (it automatically covers both of coalesced and uncoalesced cases), here only tests for case where sparseDims = 1.

Reply:
Gotcha.  Hows this?